addons/hr_holidays/models/res_users.py

```diff
@@ -36,7 +36,7 @@ def _compute_im_status(self):
         on_leave_user_ids = self._get_on_leave_ids()
         for user in self:
             if user.id in on_leave_user_ids:
-                if user.im_status == 'online':
+                if user._is_user_available():
```

Comment:
This one is the actual compute of the im_status, I don't think it should use available.

Reply:
Well it's just broken by design.
The one from hr_holidays will always conflict with hr_homeworking?

Reply:
The final `else` here might need to be "elif offline" to not do anything of presence already did stuff?

Reply:
@seb-odoo  Mmh, it feels to me like hr_holidays should have precedence over remote presence. The way it is now, going through hr_homeworking and hr_holidays computes should always give us `leave_online` no matter the order so it'd be fine, if the user has a leave then it doesn't matter where they are. I guess we'd have the same problem with offline and away as we did with online though...

Either way this can't work for both modules, so maybe the better move is not to touch hr_holidays in stable and stick to the livechat issue... I'll take your suggestion and change only the else into elif for hr_holidays